web/ui/react-app/src/pages/targets/EndpointLink.tsx

```diff
@@ -8,23 +8,23 @@ export interface EndpointLinkProps {
 
 const EndpointLink: FC<EndpointLinkProps> = ({ endpoint, globalUrl }) => {
   let url: URL;
+  let search = '';
+  let invalidURL = false;
   try {
     url = new URL(endpoint);
   } catch (err: unknown) {
-    const error = err as Error;
-    return (
-      <Alert color="danger">
-        <strong>Error:</strong> {error.message}
-      </Alert>
-    );
+    invalidURL = true;
+    if (endpoint.indexOf('?') > -1) search = endpoint.substring(endpoint.indexOf('?'));
+    url = new URL('http://0.0.0.0' + search);
   }
 
   const { host, pathname, protocol, searchParams }: URL = url;
   const params = Array.from(searchParams.entries());
-
+  const href = invalidURL ? endpoint : globalUrl;
```

Comment:
We should still use the global URL for the href regardless. Sorry if I caused confusion in https://github.com/prometheus/prometheus/issues/9760#issuecomment-977074308, I meant only for the display link.

Reply:
fixed in last commit